api/strategies/openidStrategy.js

```diff
@@ -73,6 +73,33 @@ function convertToUsername(input, defaultValue = '') {
   return defaultValue;
 }
 
+async function mapCustomOpenIdData(accessToken, customOpenIdFields) {
+  const customData = {};
+  const fieldsQueryURL = `https://graph.microsoft.com/v1.0/me?$select=${customOpenIdFields.join(',')}`;
```

Comment:
This seems unique to Microsoft OpenID services. The PR notes and documentation write-up make it seem like it could be applicable to many different OIDC providers but I don't think this is usually the case.

Different OIDC providers may have different ways of accessing additional user information:
- Google has its People API
- Auth0 has its Management API
- Okta has its User API
- etc.

1 or more of these approaches would be fine for me to merge:

1. Make it explicitly Microsoft-specific and name it accordingly (e.g., `mapMicrosoftCustomOpenIdData`)

2. Make it provider-agnostic by:
   ```javascript
   async function mapCustomOpenIdData(accessToken, customOpenIdFields, providerConfig) {
     const { userInfoEndpoint, fieldMapping } = providerConfig;
     // Use configurable endpoint and field mapping logic
     // ...
   }
   ```

3. Create an abstraction layer with provider-specific implementations:
   ```javascript
const PROVIDER_MAPPERS = {
  microsoft: MicrosoftDataMapper,
  google: GoogleDataMapper,
  // Add more providers as needed
};

class OpenIdDataMapper {
  static getMapper(provider) {
    const MapperClass = PROVIDER_MAPPERS[provider];
    if (!MapperClass) {
      throw new Error(`No mapper found for provider: ${provider}`);
    }
    return new MapperClass();
  }
}

// Example implementations
class MicrosoftDataMapper {
  async mapCustomData(accessToken, customOpenIdFields) {
    const fieldsQueryURL = `https://graph.microsoft.com/v1.0/me?$select=${customOpenIdFields.join(',')}`;
    // Microsoft-specific implementation...
  }
}

class GoogleDataMapper {
  async mapCustomData(accessToken, customOpenIdFields) {
    const fieldsQueryURL = `https://people.googleapis.com/v1/people/me...`;
    // Google-specific implementation...
  }
}
   ```

Reply:
I would prefer the 3rd approach, since that seems like it would be the cleanest and easily allow implementations for Okta, Microsoft, etc.

Reply:
Sorry for the late response. By the time I created the pull request I wasn't aware that there were other OIDC providers. You are totally right. I'll update the pull request with your comments. I also prefer the 3rd approach because it is reusable, but I have no way to test it :(. Do you have any ideas how to approach that problem ?